internal/delete_test.go

```diff
@@ -94,12 +94,37 @@ func testFindTargetBeforeTime(t *testing.T, minute int, modifier int) (storage.O
 	return internal.FindTargetBeforeTime(mockFolder, timeLine, modifier, isFullBackup, lessByTime)
 }
 
-func TestDeleteBeforeTarget_WithPermanentBackups(t *testing.T) {
-	folder := createMockStorageFolderWithPermanentBackups(t)
+func verifyThatExistBackupsAndWals(t *testing.T, expectBackupExistAfterDelete, expectWalExistAfterDelete map[string]bool, folder storage.Folder) {
 	baseBackupFolder := folder.GetSubFolder(utility.BaseBackupPath)
 	walBackupFolder := folder.GetSubFolder(utility.WalPath)
+	for backupName, expect := range expectBackupExistAfterDelete {
+		exists, err := baseBackupFolder.Exists(backupName + "/" + utility.MetadataFileName)
+		assert.NoError(t, err)
+		assert.Equal(t, expect, exists, "errored on "+backupName+"/"+utility.MetadataFileName)
+	}
+	for walName, expect := range expectWalExistAfterDelete {
+		exists, err := walBackupFolder.Exists(walName + ".lz4")
+		assert.NoError(t, err)
+		assert.Equal(t, expect, exists, "errored on "+walName+".lz4")
+	}
+}
+
+func TestDeleteBeforeTargetWithPermanentBackups(t *testing.T) {
+	folder := createMockStorageFolderWithPermanentBackups(t)
+
+	expectBackupExistBeforeDelete := map[string]bool{
+		"base_000000010000000000000002":                            true,
+		"base_000000010000000000000004_D_000000010000000000000002": true,
+		"base_000000010000000000000006_D_000000010000000000000004": true,
+	}
+	expectWalExistBeforeDelete := map[string]bool{
+		"000000010000000000000001": true,
+		"000000010000000000000002": true,
+		"000000010000000000000003": true,
+	}
+
 	expectBackupExistAfterDelete := map[string]bool{
-		"base_000000010000000000000002":                            false,
```

Comment:
WTF?

Reply:
just change 115 line. That backup should be permanent